include/SugarEmailAddress/SugarEmailAddress.php

```diff
@@ -1249,7 +1249,7 @@ public function AddUpdateEmailAddress($addr, $invalid = 0, $opt_out = 0, $id = n
 
         // confirmed opt in check
         if (!is_null($optInFlag)) {
-            $optInFlag = 0;
+            $optInFlag = (int)$optInFlag;
```

Comment:
If you convert the value to integer here, converts below this line are not necessary.